pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/AuthenticationTest.java

```diff
@@ -1,110 +1,129 @@
+    private static String cmiKitShippingOnlyUserId;
 
-        Patch patch = new Patch("0",  "dsmKitRequestId", "0", gpUserId , new NameValue("kit.collectionDate",  "2023-04-24"), null, "XSZSRS1MS3D4OAEK2DPM") ;
-        patch.setTableAlias("kit");
+    private static final UserAdminTestUtil cmiAdminUtil = new UserAdminTestUtil();
+
+    private static String studyInstanceName;
 
-        Assert.assertTrue(UserUtil.checkKitShippingAccessForPatch(realm, gpUserId, null, patch));
+    private static String generateUserEmail() {
+        return "AuthTest-" + System.currentTimeMillis() + "@broad.dev";
+    }
+
+    /**
+     * Create new groups, a new ddp_instance, associated roles,
+     * a few new users and set their permissions.
+     */
+    @BeforeClass
+    public static void setup() {
+        String nameAppend = "." + System.currentTimeMillis();
+        String studyInstanceName = "instance." + nameAppend;
+        String cmiStudyGroup = "cmi" + nameAppend;
+
+        cmiAdminUtil.createRealmAndStudyGroup(studyInstanceName, cmiStudyGroup);
+        cmiAdminUtil.setStudyAdminAndRoles(generateUserEmail(), USER_ADMIN_ROLE, Arrays.asList(KIT_SHIPPING, PT_LIST_VIEW));
+
+        dsmAdminUserId = Integer.toString(cmiAdminUtil.createTestUser(generateUserEmail(), Arrays.asList(KIT_SHIPPING, PT_LIST_VIEW)));
+        cmiKitShippingOnlyUserId = Integer.toString(cmiAdminUtil.createTestUser(generateUserEmail(), Collections.singletonList(KIT_SHIPPING)));
     }
 
     @Test
-    @Ignore
-    public void GPPatchOtherAccessTest(){
-        String realm = "osteo2";
+    public void testKitShipperRoleCanChangeKitInfo() {
+        Patch patch = new Patch("0",  "dsmKitRequestId", "0", cmiKitShippingOnlyUserId , new NameValue("kit.collectionDate",  "2023-04-24"), null, "XSZSRS1MS3D4OAEK2DPM") ;
+        patch.setTableAlias("kit");
 
-        String gpUserId = Integer.toString(new UserDao().getUserByEmail(cfg.getString(GP_UNIT_TESTER_EMAIL)).orElse(new UserDto()).getId());
+        Assert.assertTrue(UserUtil.checkKitShippingAccessForPatch(studyInstanceName, cmiKitShippingOnlyUserId, null, patch));
+    }
 
-        Patch patch = new Patch("0", "participantId", "0", gpUserId , new NameValue("oD.locationPx",  "location"), null, "XSZSRS1MS3D4OAEK2DPM") ;
+    @Test
+    public void getKitShipperRoleCannotChangeParticipantInfo(){
+        Patch patch = new Patch("0", "participantId", "0", cmiKitShippingOnlyUserId, new NameValue("oD.locationPx",  "location"), null, "XSZSRS1MS3D4OAEK2DPM") ;
         patch.setTableAlias("oD");
 
-        Assert.assertFalse(UserUtil.checkKitShippingAccessForPatch(realm, gpUserId, null, patch));
+        Assert.assertFalse(UserUtil.checkKitShippingAccessForPatch(studyInstanceName, cmiKitShippingOnlyUserId, null, patch));
     }
 
+    /**
+     * Verifies that users who don't have permission to make
```

Comment:
This seems a bit different. We get a user ID from AuthO and there is also one in the payload. (Not sure why we do things this way.) This verifies that when the two don't match we return an error?

Reply:
CC @pegahtah since I'm not familiar with the rationale behind this test.

Reply:
Yes! The user id in the payload should match the one in the token. So basically _I think_ the idea behind this is that I can't use someone else's token to change a value in DSM.
 The user id in the request payload is going to get used for the audit purposes